### PR TITLE
Return sindy coefficients

### PR DESCRIPTION
**Pull request by AlCap23**
For reuse

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Why not have them stored in the Basis in a way that's easier to get?

**Comment by AlCap23** [comment]
I've thought about that. And right now this is just a way to get them out of the function call.

But I was wondering about the usefulness of this approach, since most of the time I am not interested in the coefficient matrix.

The best solution in my opinion would be to add the coefficients as parameters  and append this to the parameters of the basis. This would allow an easy transition into the parameter estimation / diffeqflux framework, wouldn't it?

**Comment by ChrisRackauckas** [comment]
Or two functions: one for getting the SInDy coeffficients, then building a Basis from it, and the standard SInDy call does both together. But a quick refactor would let you grab the process separately from the constructor. Putting an if statement on here seems like the wrong idea.

File: test/runtests.jl [hunk 1]
```diff
@@ -243,8 +238,8 @@ end
 
     # Check with noise
     X = sol[:, :] + 1e-3*randn(size(sol[:,:])...)
-    set_threshold!(opt, 3.5e-1)
-    Ψ = SInDy(X, DX, basis, maxiter = 10000, opt = opt, denoise = true, normalize = true)
+    #set_threshold!(opt, 3.5e-1)
+    Ψ = SInDy(X, DX, basis, λs, maxiter = 10000, opt = opt, denoise = true, normalize = true)
```
Comment (ChrisRackauckas) [inline test/runtests.jl]: is `sparse_regression` going to be considered user-level API? If so, the user-side usage should probably get a test.

Comment (AlCap23) [reply test/runtests.jl]: Yes, working on this. Also to be consistent add the same for implicit sindy.

**Comment by AlCap23** [comment]
Did the change to `SciML/DataDrivenDiffEq.jl` just happen within the last test ? :)

**Pull request merged**